using-mobile-extensions/adobe-journey-optimizer-decisioning/api-reference.md

```diff
@@ -0,0 +1,542 @@
+```
+
+{% endtab %}
+{% endtabs %}
+
+## updatePropositions
+
+This API dispatches an Event for the Edge network extension to fetch decision propositions, for the provided decision scopes array, from the decisioning services enabled in the Experience Edge. The returned decision propositions are cached in-memory in the Optimize SDK extension and can be retrieved using `getPropositions` API.
+
+{% tabs %}
+{% tab title="iOS — Swift" %}
+### Swift
+
+#### Syntax
+
+```swift
+static func updatePropositions(for decisionScopes: [DecisionScope], 
+                               withXdm xdm: [String: Any]?,
+                               andData data: [String: Any]? = nil)
+```
+
+* _decisionScopes_ is an array of decision scopes for which propositions need updating.
+* _xdm_ additional xdm formatted data to be attached to the Experience Event.
+* _data_ additional freeform data to be attached to the Experience Event.
+
+#### Example
+
+```swift
+let decisionScope1 = DecisionScope(activityId: "xcore:offer-activity:1111111111111111", 
+                                   placementId: "xcore:offer-placement:1111111111111111" 
+                                   itemCount: 2)
+let decisionScope2 = DecisionScope(name: "myScope")
+
+Optimize.updatePropositions(for: [decisionScope1, decisionScope2] 
+                            withXdm: ["xdmKey": "xdmValue"] 
+                            andData: ["dataKey": "dataValue"])
+```
+
+### Objective-C
+
+#### Syntax
+
+```objc
++ (void) updatePropositions: (NSArray<AEPDecisionScope*>* _Nonnull) decisionScopes 
+                    withXdm: (NSDictionary<NSString*, id>* _Nullable) xdm
+                    andData: (NSDictionary<NSString*, id>* _Nullable) data;
+```
+
+#### Example
+
+```objc
+AEPDecisionScope* decisionScope1 = [[AEPDecisionScope alloc] initWithActivityId: @"xcore:offer-activity:1111111111111111" 
+                                                                   placementId: @"xcore:offer-placement:1111111111111111" 
+                                                                     itemCount: 2];
+AEPDecisionScope* decisionScope2 = [[AEPDecisionScope alloc] initWithName: @"myScope"]; 
+
+[AEPMobileOptimize updatePropositions: @[decisionScope1, decisionScope2] 
+                              withXdm: @{@"xdmKey": @"xdmValue"} 
+                              andData: @{@"dataKey": @"dataValue"}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Public classes
+
+| Type | Swift | Objective-C |
+| :--- | :--- | :--- |
+| class | `DecisionScope` | `AEPDecisionScope` |
+| class | `Proposition` | `AEPProposition` |
+| class | `Offer` | `AEPOffer` |
+
+{% tabs %}
+{% tab title="iOS — Swift" %}
+### DecisionScope
+
+This class represents the decision scope which is used to fetch the decision propositions from the Edge decisioning services. The encapsulated scope name can also represent the Base64 encoded JSON string created using the provided activityId, placementId and itemCount.
```

Comment:
"... represent the Base64-encoded JSON string... provided activityId, placementId, and itemCount.".